frontend/src/views/Viewlevelstable.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <div class="PageWrapper">
+    <div class="TableContainer">
+      <h2>Attached Levels for Role ID: {{ roleId }}</h2>
+      <br />
+      <ElTable :data="levels" style="width: 100%">
+        <ElTableColumn prop="name" label="Levels" />
+      </ElTable>
+      <div class="NewLevel">
+        <ElButton type="primary" @click="redirectToLevelView">Create Level</ElButton>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, watch } from 'vue'
+import { ElTable, ElTableColumn, ElButton } from 'element-plus'
+import { useRouter, useRoute } from 'vue-router'
+
+const router = useRouter()
+const route = useRoute()
+
+const roleId = ref(route.params.roleId ? String(route.params.roleId) : '')
+
+const levels = ref<Array<{ id: number; name: string; role_id: number }>>([])
```

Comment:
Instead of doing this, use (or create) a Level class
```suggestion
const levels = ref<Array<Level>>([])
```